src/auth/guards/driver.guard.ts

```diff
@@ -0,0 +1,18 @@
+// driver.guard.ts
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { AuthRoles } from 'src/common/enums/roles.enum';
+import { AuthErrors } from 'src/common/error/auth.error';
+
```

Comment:
This is a bad solution. If we have 10 roles you will create 10 guards? Thinks about this.